### PR TITLE
updated outdated SecurityDirectives documentation

### PR DESCRIPTION
**Pull request by Prithvirajbilla**
Fixes akka/akka-http#1715

### CODE REVIEW

**Comment by jlprat** [comment]
OK TO TEST

File: docs/src/main/paradox/routing-dsl/directives/security-directives/authenticateBasic.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ Longer-running authentication tasks (like looking up credentials in a database)
 variant of this directive which allows it to run without blocking routing layer of Akka HTTP, freeing it for other requests.
 
 Standard HTTP-based authentication which uses the `WWW-Authenticate` header containing challenge data and
-@unidoc[Authorization] header for receiving credentials is implemented in subclasses of `HttpAuthenticator`.
+@unidoc[Authorization] header for receiving credentials is implemented in `extractCredentialsAndAuthenticateOrRejectWithChallenge`.
```
Comment (johanandren) [inline docs/src/main/paradox/routing-dsl/directives/security-directives/authenticateBasic.md]: That is an internal method, it should be `authenticateOrRejectWithChallenge` which is the public API directive

Comment (Prithvirajbilla) [reply docs/src/main/paradox/routing-dsl/directives/security-directives/authenticateBasic.md]: my bad! I didn't realize that. updated now.

**Comment by Prithvirajbilla** [comment]
@johanandren can you review this new PR https://github.com/akka/akka-http/pull/2102. sorry, I didn't realize force sync master closes the PR.

**Pull request closed without merging**